src/api/libcellml/annotator.h

```diff
@@ -0,0 +1,289 @@
+     * Factory method to create a @c Annotator.  Create a
+     * annotator with::
+     *
+     *   AnnotatorPtr annotator = libcellml::Annotator::create();
+     *
+     * @return A smart pointer to an @c Annotator object.
+     */
+    static AnnotatorPtr create() noexcept;
+
+    /**
+     * @brief Builds the internal map of ids to items for the model provided.
+     *
+     * @param model A @c ModelPtr model to build map for.
+     */
+    void build(const ModelPtr &model);
+
+    /**
+     * @brief Retrieves an item with the given id string, in the model
+     * for which the internal map was built using @sa build().
+     *
+     * The item returned is a @c std::pair containing:
+     *  - an @c Annotator::Type enum, and
+     *  - an @c std::any item containing the item.
+     *
+     * Possible string labels and their corresponding items returned are:
+     *  - Annotator::Type::COMPONENT
+     *          ComponentPtr to item with this id.
+     *  - Annotator::Type::COMPONENT_REF
+     *          ComponentPtr to component which has component->componentRefId() == id.
+     *  - Annotator::Type::CONNECTION
+     *          VariablePair including two VariablePtr items which can be used to retrieve the
+     *          connection.  Note that multiple pairs of variables could be used to identify
+     *          a single connection.
+     *   Annotator::Type::ENCAPSULATION
+     *          An empty string.
+     *   Annotator::Type::IMPORT
+     *          An ImportSourcePtr item with this id.
+     *   Annotator::Type::ISSUE
+     *          An IssuePtr containing an error message.
+     *   Annotator::Type::MAP_VARIABLES
+     *          A VariablePair including two VariablePtr items which can be used to retrieve the
+     *          equivalence.
+     *   Annotator::Type::MODEL
+     *          A ModelPtr with this id.
+     *   Annotator::Type::RESET
+     *          A ResetPtr with this id.
+     *   Annotator::Type::RESET_VALUE
+     *          A ResetPtr which is the parent of the reset_value item found with this id.
+     *          Retrieve the reset_value MathML string using reset->resetValue().
+     *   Annotator::Type::TEST_VALUE
+     *          A ResetPtr which is the parent of the test_value item found with this id.
+     *          Retrieve the test_value MathML string using reset->testValue().
+     *   Annotator::Type::UNIT
+     *          A UnitItem pair containing:
+     *              .first: A UnitsPtr to the parent of the unit item with this id, and
+     *              .second: A size_t with the index to the Unit item.
+     *          Retrieve the unit description using Units::unitAttributes() function with
+     *   Annotator::Type::UNITS
+     *          A UnitsPtr item with this id.
+     *   Annotator::Type::VARIABLE
+     *          A VariablePtr item with this id.
+          *
+     * @param id A @c std::string representing the @p id to retrieve.
+     *
+     * @return An @c AnyItem item as described above.
```

Comment:
Not at all endorsing the "return an issue type AnyItem on error" - but if that remains it should be clear in the docs here that is what would happen.